Physicist/Physicist/Actor/Actor.cs

```diff
@@ -297,5 +342,79 @@ protected virtual bool OnCollision(Fixture fixtureA, Fixture fixtureB, FarseerPh
 
             return this.IsEnabled;
         }
+
+        protected void PlaySound(string name)
+        {
+            List<string> removelist = new List<string>();
+
+            // check for every Guid in the sound controller.
+            for (int i = 0; i < this.PlayingSounds.Count; i++)
+            {
+                if (!SoundController.ContainsInstance(this.PlayingSounds.ElementAt(i).Value))
+                {
+                    // add to a removelist
+                    removelist.Add(this.PlayingSounds.ElementAt(i).Key);
+                }
+            }
+
+            // remove everything in the removelist
+            foreach (string s in removelist)
+            {
+                this.PlayingSounds.Remove(s);
+            }
+
+            Guid newGuid;
+            newGuid = SoundController.PlayLocalizedSound(name, this.audioEmitter);
+
+            if (this.PlayingSounds.ContainsKey(name))
+            {
+                this.PlayingSounds[name] = newGuid;
+            }
+            else
+            {
+                this.PlayingSounds.Add(name, newGuid);
+            }
+        }
+
+        protected void PlaySound(string name, bool localized, bool looping, float volume, float pitch, float pan)
+        {
+            List<string> removelist = new List<string>();
+
+            // check for every Guid in the sound controller.
+            for (int i = 0; i < this.PlayingSounds.Count; i++)
+            {
+                if (!SoundController.ContainsInstance(this.PlayingSounds.ElementAt(i).Value))
+                {
+                    // add to a removelist
+                    removelist.Add(this.PlayingSounds.ElementAt(i).Key);
+                }
+            }
+
+            // remove everything in the removelist
+            foreach (string s in removelist)
+            {
+                this.PlayingSounds.Remove(s);
+            }
+
+            Guid newGuid;
+
+            if (localized)
+            {
+                newGuid = SoundController.PlayLocalizedSound(name, this.audioEmitter, looping, volume, pitch, pan);
+            }
+            else
+            {
+                newGuid = SoundController.PlaySound(name, looping, volume, pitch, pan);
+            }
+
+            if (this.PlayingSounds.ContainsKey(name))
+            {
+                this.PlayingSounds[name] = newGuid;
```

Comment:
We discussed this before.  It will make it only possible to pause/resume/stop the last copy of the sound playing *per Actor*... the idea being that longer or looping sounds shouldn't really need to have multiples playing on a single actor.  It works fine if these long sounds are, say, playing on different actors.  I suppose this could be reinforced by disallowing the playing of looped sounds if a version of that sound is already active in this.PlayingSounds.

Since the SoundController stores the instances by Guid, however, we can have as many copies from different actors as we want.